subprojects/base-services/src/main/java/org/gradle/internal/progress/BuildOperationDescriptor.java

```diff
@@ -21,21 +21,27 @@
 /**
  * Meta-data about a build operation.
  */
-public class BuildOperationDetails {
-    private final BuildOperationExecutor.Operation parent;
+public class BuildOperationDescriptor {
+    private final Long id;
```

Comment:
We should move `OperationIdentifier` over to the build operation package and use it there. I think having an ID class is a good idea instead of using a plain long, as it makes it less likely to accidentally compare apples to oranges.